SampleCode/Python/sample_identify_cross_page_tables.py

```diff
@@ -161,32 +326,100 @@ def identify_cross_page_tables(input_file_path):
+            and table_concision_list[pre_table_idx + 1]["min_offset"]
+            - table_concision_list[pre_table_idx]["max_offset"]
+            <= SEPARATOR_LENGTH_IN_MARKDOWN_FORMAT
+        )
+
+        if is_vertical or is_horizontal:
+            print(f"Merge table: {pre_table_idx} and {pre_table_idx + 1}")
             print("----------------------------------------")
 
+            remark = ""
+            cur_content = result.content[table_concision_list[pre_table_idx + 1]["min_offset"] : table_concision_list[pre_table_idx + 1]["max_offset"]]
+
+            if is_horizontal:
+                    remark = result.content[table_concision_list[pre_table_idx]["max_offset"] : table_concision_list[pre_table_idx + 1]["min_offset"]]
+            
+            merged_list_len = len(merged_table_list)
+            if merged_list_len > 0 and len(merged_table_list[-1]["table_idx_list"]) > 0 and merged_table_list[-1]["table_idx_list"][-1] == pre_table_idx:
+                merged_table_list[-1]["table_idx_list"].append(pre_table_idx + 1)
+                merged_table_list[-1]["offset"]["max_offset"]= table_concision_list[pre_table_idx + 1]["max_offset"]
+                if is_vertical:
+                    merged_table_list[-1]["content"] = merge_vertical_tables(merged_table_list[-1]["content"], cur_content)
+                elif is_horizontal:
+                    merged_table_list[-1]["content"] = merge_horizontal_tables(merged_table_list[-1]["content"], cur_content)
+                    merged_table_list[-1]["remark"] += remark
+
+            else:
+                pre_content = result.content[table_concision_list[pre_table_idx]["min_offset"] : table_concision_list[pre_table_idx]["max_offset"]]
+                merged_table = {
+                    "table_idx_list": [pre_table_idx, pre_table_idx + 1],
+                    "offset": {
+                        "min_offset": table_concision_list[pre_table_idx]["min_offset"],
+                        "max_offset": table_concision_list[pre_table_idx + 1]["max_offset"],
+                        },
+                    "content": merge_vertical_tables(pre_content, cur_content) if is_vertical else merge_horizontal_tables(pre_content, cur_content),
+                    "remark": remark.strip() if is_horizontal else ""
+                    }
+                
+                if merged_list_len <= 0:
+                    merged_table_list = [merged_table]
+                else:
+                    merged_table_list.append(merged_table)
+
+    
+    optimized_content= ""
+    if merged_table_list:
+        print(f"Merged {len(merged_table_list)} markdown tables totally.")
+        print("=========================================================")
+        start_idx = 0
+        for merged_idx, merged_table in enumerate(merged_table_list):
+            print(f"Hight the merged table {merged_idx}")
+            print("-----------------------------------------------------")
```

Comment:
Highlight?

Reply:
the merged idx is not very straight forward - can we reference the original table index so we know which ones are merged?

Reply:
“Highlight” is inappropriate.  Refactored it follow the suggestion. Thx